src/pybullet_blocks/planning_models/action.py

```diff
@@ -403,13 +404,17 @@ def _generate_table_placements(
             raise NotImplementedError
 
     def _generate_block_placements(
-        self, held_obj_id: int, target_id: int, state: KinematicState
+        self, held_obj_id: int, target_id: int, _state: KinematicState
     ) -> Iterator[Pose]:
         held_obj_half_height = self._sim.get_object_half_extents(held_obj_id)[2]
         target_half_height = self._sim.get_object_half_extents(target_id)[2]
         dz = target_half_height + held_obj_half_height
-        target_orientation = state.object_poses[target_id].orientation
-        yield Pose((0, 0, dz), target_orientation)
+
+        # Sample rotated orientations
+        while True:
+            yaw = np.random.uniform(-np.pi, np.pi)
```

Comment:
I just want to confirm that you want to permit any yaw. Alternatively, you could only sample 90 degree increments. Up to you--it's mostly an aesthetic choice.

Reply:
coincidentally I just needed to do a similar thing for another project, for your reference: https://github.com/tomsilver/pybullet-helpers/pull/106/files